dao-contracts/tests/test_repo_voter.rs

```diff
@@ -0,0 +1,174 @@
+    let voting: Voting = repo_voter_contract.get_voting(voting_id);
+
+    // vote to reach quorum
+    repo_voter_contract
+        .as_account(env.get_account(1))
+        .vote(voting_id, true, 1000.into())
+        .unwrap();
+    repo_voter_contract
+        .as_account(env.get_account(2))
+        .vote(voting_id, false, 1000.into())
+        .unwrap();
+
+    // advance time, so voting can be finished
+    env.advance_block_time_by(Duration::from_secs(voting.formal_voting_time() + 1));
+
+    // finish voting
+    repo_voter_contract.finish_voting(voting_id).unwrap();
+
+    // the action should be performed
+    let bytes = variable_repo_contract.get("variable_name".into()).unwrap();
+    let (variable, bytes) = String::from_bytes(&bytes).unwrap();
+    assert_eq!(bytes.len(), 0);
+    assert_eq!(variable, "new_value");
+
+    // those who voted against' reputation should be transferred to for voters proportionally
+    assert_eq!(
+        reputation_token_contract.balance_of(repo_voter_contract.address()),
+        1.into()
+    );
+    assert_eq!(
+        reputation_token_contract.balance_of(env.get_account(0)),
+        (10000 + 333).into()
+    );
+    assert_eq!(
+        reputation_token_contract.balance_of(env.get_account(1)),
+        (10000 + 666).into()
+    );
+    assert_eq!(
+        reputation_token_contract.balance_of(env.get_account(2)),
+        (10000 - 1000).into()
+    );
+
+    // as the reputation was not divisible entirely, we check the dust amount
+    assert_eq!(repo_voter_contract.get_dust_amount(), 1.into());
+}
+
+pub fn setup() -> (
+    TestEnv,
+    RepoVoterContractTest,
+    VariableRepositoryContractTest,
+    ReputationContractTest,
+) {
+    let env = TestEnv::new();
+    let mut variable_repo_contract = VariableRepositoryContractTest::new(&env);
+    let mut reputation_token_contract = ReputationContractTest::new(&env);
+    let mut repo_voter_contract = RepoVoterContractTest::new(
+        &env,
+        Address::from(variable_repo_contract.get_package_hash()),
+        Address::from(reputation_token_contract.get_package_hash()),
+    );
+
+    variable_repo_contract
+        .add_to_whitelist(repo_voter_contract.address())
+        .unwrap();
+
+    // TODO: Currently there are 4 hardcoded onboarded members, quorum at 750 should make the quorum when there are 3 votes
+    variable_repo_contract
+        .update_at(
+            consts::INFORMAL_VOTING_QUORUM.into(),
+            U256::from(750).to_bytes().unwrap().into(),
+            None,
+        )
+        .unwrap();
+    variable_repo_contract
+        .update_at(
+            consts::FORMAL_VOTING_QUORUM.into(),
+            U256::from(750).to_bytes().unwrap().into(),
+            None,
+        )
+        .unwrap();
+
+    reputation_token_contract
+        .add_to_whitelist(repo_voter_contract.address())
+        .unwrap();
+
+    reputation_token_contract
+        .mint(env.get_account(0), 10000.into())
```

Comment:
In many places, the same values are used, and I believe their meaning is the same. I would consider defining some consts on the file beginning.